code/weapon/weapons.cpp

```diff
@@ -5632,8 +5755,59 @@ void weapon_home(object *obj, int num, float frame_time)
 		vm_vec_copy_scale( &obj->phys_info.desired_vel, &obj->orient.vec.fvec, obj->phys_info.speed);
 
 		vec3d turnrate_mod = vm_vec_new(1.0f, 1.0f, 1.0f);
-		if (wip->turn_rate_curve_idx >= 0)
-			turnrate_mod *= Curves[wip->turn_rate_curve_idx].GetValue(time_alive / wip->lifetime);
+
+		for (uint c = 0; c < wip->lifetime_curves.size(); c++) {
+			WeaponLifetimeCurve* mod_curve = &wip->lifetime_curves[c];
+			if (mod_curve->curve_idx < 0) {
+				Warning(LOCATION, "Curve does not exist!");
+				continue;
+			}
+
+			Curve curve = Curves[mod_curve->curve_idx];
+			float input = 1.0f;
+			float output = 1.0f;
+
+			// we do the modular curve processing in this sort of inside-out way because we only care about one output
+			// so we don't want to bother doing any processing for curves with other outputs
+			switch (mod_curve->output) {
+				case WeaponLifetimeCurveOutput::TURN_RATE_MULT: {
```

Comment:
Might be a bit easier to have `if (mod_curve->output == WeaponLifetimeCurveOutput::TURN_RATE_MULT)` ? Ditto for det radius